console-subscriber/src/lib.rs

```diff
@@ -494,6 +500,17 @@ impl<S> Layer<S> for ConsoleLayer
 where
     S: Subscriber + for<'a> LookupSpan<'a>,
 {
+    fn on_register_dispatch(&self, dispatch: &Dispatch) {
+        let tracer = consequences::Tracer::from_dispatch::<S>(dispatch);
+        if let Err(_) = self.shared.tracer.set(tracer) {
+            // XXX(jswrenn): This layer has seemingly been supplied *multiple*
+            // subscribers. Is that possible? I guess you can call
+            // `Layer::with_subscriber` multiple times; what happens if you
+            // install the resulting `Layered` as a subscriber? Is `Arc`ing a
+            // `Layered<ConsoleLayer, S>` also a recipe for weirdness?
+        }
+    }
```

Comment:
The `on_register_dispatch` method defined by https://github.com/tokio-rs/tracing/pull/2269 is implemented here, to construct a `Tracer` (i.e., a type that defines a `trace` method that consumes a span ID and produces a trace of its consequences), which is then made available to the `Aggregator` thread (the part of console_subscriber that actually handles the rpcs for reporting traces).